src/compiler/moduleNameResolver.ts

```diff
@@ -2292,8 +2291,8 @@ function loadEntrypointsFromExportMap(
                     /*excludes*/ undefined,
                     [
                         isDeclarationFileName(target)
-                            ? replaceFirstStar(target, "**/*")
-                            : changeAnyExtension(replaceFirstStar(target, "**/*"), getDeclarationEmitExtensionForPath(target)),
+                            ? changeAnyExtension(replaceFirstStar(target, "**/*"), ".*", supportedDeclarationExtensions, /*ignoreCase*/ true)

```

Comment:
It seems like we are only filtering out replacing to `*.*` only if extension is of type ".d.someext.ts"
Was that intentional. I  wonder if we can just always do `changeAnyExtension(replaceFirstStar(target, "**/*"), ".*")` irrespective of extension .

Reply:
The problem is `changeAnyExtension(path, ".*")` only replaces from the last `.`, so that would result in `foo.d.ts` becoming `foo.d.*`, when we need it to become `foo.*` instead.

Reply:
but if i locally change this to do ` changeAnyExtension(replaceFirstStar(target, "**/*"), ".*"),` none of the tests fail ?

Reply:
You're right. There's currently no test where the `exports` target has a declaration file extension but we find a `.ts` on disk instead. While that should arguably work, it's not very realistic.

Reply:
Presumably we don't have many tests with export maps pointing explicitly at declaration files with js or implementation files alongside them that we'd prefer to lookup - it's usually the other way around, so only finding matches for `name.d.*` gets us what we want almost all the time.

Reply:
Yeah, exactly. Still, I running with an include of `name.d.*` feels too much like a mistake to me. I made a new helper `changeFullExtension` that recognizes and swaps all declaration file extensions (including ones for arbitrary file extensions, which the current code doesn't handle correctly), and am just trying to figure out how to push the commit I made on my host machine from my dev box 😅